benchmarks/graphics_pipeline/main.cpp

```diff
@@ -783,6 +841,37 @@ void ProjApp::Setup()
     }
 }
 
+void ProjApp::KeyDown(ppx::KeyCode key)
+{
+    mPressedKeys[key] = true;
+}
+
+void ProjApp::KeyUp(ppx::KeyCode key)
+{
+    mPressedKeys[key] = false;
+}
+
+void ProjApp::ProcessInput()
+{
+    float deltaTime = GetPrevFrameTime();
+
+    if (mPressedKeys[KEY_W]) {
+        mCamera.Move(FreeCamera::MovementDirection::FORWARD, kCameraSpeed * deltaTime);
```

Comment:
I think it's fine as the first iteration, but there might be some weird edge cases here. What if the application is really slow and both KeyDown and KeyUp are called in between input processing events? Also this doesn't actually take into account when the duration of the key press.